addons/htpasswd-identity-provider/htpasswd-identity-provider.addon

```diff
@@ -1,30 +1,45 @@
 # Name: htpasswd-identity-provider
 # Description: Configures minishift to use HTPasswdIdentityProvider
 # Url: https://docs.okd.io/3.9/install_config/configuring_authentication.html#HTPasswdPasswordIdentityProvider
-# Var-Defaults: USERNAME=developer,USER_PASSWORD=developer,MINISHIFT_DATA_HOME=/var/lib/minishift,CONFIG_LOCATION=openshift.local.config/master,MASTER_CONFIG_FILE=master-config.yaml,ORIGIN_DATA_HOME=/var/lib/origin
-# Required-Vars: USER_PASSWORD,USERNAME,CONFIG_LOCATION
+# Var-Defaults: USERNAME=developer,USER_PASSWORD=developer,MINISHIFT_DATA_HOME=/var/lib/minishift,CONFIG_LOCATION_OPENSHIFT_API=base/openshift-apiserver,CONFIG_LOCATION_KUBERNETES_API=base/kube-apiserver,MASTER_CONFIG_FILE=master-config.yaml
+# Required-Vars: USER_PASSWORD,USERNAME,CONFIG_LOCATION_OPENSHIFT_API
+# OpenShift-Version: >=3.10.0
 
 # Install htpasswd on the origin container
 docker exec -t origin /usr/bin/bash -c "which htpasswd || yum install -y httpd-tools"
 
 # backup the existing master-config.yaml
-ssh sudo cp -fp #{MINISHIFT_DATA_HOME}/#{CONFIG_LOCATION}/#{MASTER_CONFIG_FILE} #{MINISHIFT_DATA_HOME}/#{CONFIG_LOCATION}/master-config-htpasswd.yaml
+ssh sudo cp -fp #{MINISHIFT_DATA_HOME}/#{CONFIG_LOCATION_OPENSHIFT_API}/#{MASTER_CONFIG_FILE} #{MINISHIFT_DATA_HOME}/#{CONFIG_LOCATION_OPENSHIFT_API}/master-config-htpasswd.yaml
+ssh sudo cp -fp #{MINISHIFT_DATA_HOME}/#{CONFIG_LOCATION_KUBERNETES_API}/#{MASTER_CONFIG_FILE} #{MINISHIFT_DATA_HOME}/#{CONFIG_LOCATION_KUBERNETES_API}/master-config-htpasswd.yaml
 
 # create users.htpasswd file 
-ssh sudo touch #{MINISHIFT_DATA_HOME}/#{CONFIG_LOCATION}/users.htpasswd
+ssh sudo touch #{MINISHIFT_DATA_HOME}/#{CONFIG_LOCATION_OPENSHIFT_API}/users.htpasswd
+ssh sudo touch #{MINISHIFT_DATA_HOME}/#{CONFIG_LOCATION_KUBERNETES_API}/users.htpasswd
 
-# add the default developer user to file
-docker exec -t origin /usr/bin/bash -c "htpasswd -b #{ORIGIN_DATA_HOME}/#{CONFIG_LOCATION}/users.htpasswd #{USERNAME} #{USER_PASSWORD}"
+# copy htpasswd from container to VM
+ssh sudo docker cp origin:/usr/bin/htpasswd /usr/local/bin/
 
-# Patch the master configuration to use HTPasswdPasswordIdentityProvider
-ssh sudo grep "HTPasswdPasswordIdentityProvider" #{MINISHIFT_DATA_HOME}/#{CONFIG_LOCATION}/#{MASTER_CONFIG_FILE} > /dev/null || sudo #{MINISHIFT_DATA_HOME}/bin/oc ex config patch #{MINISHIFT_DATA_HOME}/#{CONFIG_LOCATION}/master-config-htpasswd.yaml --patch='{"oauthConfig": {"identityProviders": [ {"challenge": true,"login": true,"mappingMethod": "add","name": "htpasswd","provider": {"apiVersion": "v1","kind": "HTPasswdPasswordIdentityProvider","file": "users.htpasswd"}}]}}' > #{MINISHIFT_DATA_HOME}/#{CONFIG_LOCATION}/#{MASTER_CONFIG_FILE}
+# Add user to users.htpasswd file
+ssh sudo htpasswd -b #{MINISHIFT_DATA_HOME}/#{CONFIG_LOCATION_OPENSHIFT_API}/users.htpasswd #{USERNAME} #{USER_PASSWORD}
```

Comment:
can we invoke `htpasswd` using an absolute path for now (and place it somewhere else)?